### DISCUSSION TITLE
Media elements require special handling in `:remove()`

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#3195] [General]
Because [by the spec] they are supposed to play the audio even after removed from the DOM

>[Media elements](https://html.spec.whatwg.org/multipage/media.html#media-element) that are [potentially playing](https://html.spec.whatwg.org/multipage/media.html#potentially-playing) while not [in a document](https://dom.spec.whatwg.org/#in-a-document) must not play any video, but should play any audio component. Media elements must not stop playing just because all references to them have been removed; only once a media element is in a state where no further audio could ever be played by that element may the element be garbage collected.

https://stackoverflow.com/questions/50937026/why-does-a-video-stop-playing-when-removed-from-the-dom-while-it-can-still-pl

For: https://www.reddit.com/r/uBlockOrigin/comments/1bs7vd6/im_having_a_war_with_the_twitch_front_page/

**Comment by Yuki2718** [comment]
I thought if a video autoplays it must be muted. But maybe `$permission`?

**Reply by ghost** [reply]
After interaction with a page, video can autoplay with sound.

***

Or a case when player is moved on page (miniplayer) - desired when displayed as main, not desired minimized to the corner of page.
